cinema/models.py

```diff
@@ -12,40 +16,49 @@ class CinemaHall(models.Model):
     def capacity(self) -> int:
         return self.rows * self.seats_in_row
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.name
 
 
 class Genre(models.Model):
     name = models.CharField(max_length=255, unique=True)
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.name
 
 
 class Actor(models.Model):
     first_name = models.CharField(max_length=255)
     last_name = models.CharField(max_length=255)
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.first_name + " " + self.last_name
 
     @property
-    def full_name(self):
+    def full_name(self) -> str:
         return f"{self.first_name} {self.last_name}"
 
 
+def movie_image_file_path(movie, filename):
+    _, ext = os.path.splitext(filename)
+
+    filename = f"{slugify(movie.title)}-{uuid.uuid4()}{ext}"
```

Comment:
I think it's better in terms of readability to avoid calling functions or performing operations inside an f-string. But it's up to you to decide if you want to change it, I guess.